app/controllers/reservations_controller.rb

```diff
@@ -243,6 +243,12 @@ def switch_instrument
       when params[:switch] == "off"
         switch_instrument_off!
       end
+    rescue AASM::InvalidTransition => e
+      if e.failures.include?(:time_data_completeable?)
+        flash[:error] = text("switch_instrument.prior_is_still_running")
+      else
+        raise
```

Comment:
This will cause it to fall back to the default `rescue => e` a couple lines down.

Reply:
Yeah, that's a good call. I like that better; definitely clearer.